workbench/src/renderer/components/DataDownloadModal/index.jsx

```diff
@@ -45,7 +63,16 @@ class DataDownloadModal extends React.Component {
   }
 
   async componentDidMount() {
-    const registry = JSON.parse(JSON.stringify(sampledataRegistry));
+
+    const { investList } = this.props;
+    const registry = {};
+    for (const modelName of Object.keys(investList)) {
+      let spec = await investGetSpec(investList[modelName].model_name)
+      if (spec.uiSpec.sampledata) {
+        registry[modelName] = spec.uiSpec.sampledata;
+      }
+    }
```

Comment:
This `await` inside a `for` loop raised my eyebrows a bit so I decided to profile it against an approach that uses `Promise.all` instead:

```javascript
    const specs = await Promise.all(
      Object.keys(investList).map(
        async (modelName) => await investGetSpec(investList[modelName].model_name)
      )
    );
```

The results are really interesting and the `for` loop approach clearly wins!

Here's the profile of the `for` loop:
![for_await](https://github.com/natcap/invest/assets/4071255/b49b3b74-7e3d-46ee-b853-572d95caa7e8)

All the requests are clearly happening synchronously, which is what I was concerned about. 

The `Promise.all` approach can fire them all off in parallel and wait until they are all finished to proceed. But that turns out to be way slower, probably because our backend is not designed to meet that demand:
![Promise_all](https://github.com/natcap/invest/assets/4071255/2297dc70-b88c-4b4e-8cad-3ba8646c52fa)

Lots of parallel requests here, but here's a legend for each one of those horizontal bars:
⚪ white = "queueing"
⚫ gray = "stalled"
 _"DNS lookup", "initial connection", and "request sent" all happen here very quickly_
🟢 green = "waiting for server response"
🔵 blue = "content download"

Reply:
thanks for investigating that, @davemfish! interesting, my linter is flagging that block of code for using `for... of` syntax, but it seems it's not that simple.

Reply:
I moved the sampledata registry back to where it was before as we discussed on the team call, so the `getspec` requests aren't happening any more.